### PR TITLE
Add a new validation to Catlin to warn about use of params in script blocks

### PR DESCRIPTION
**Pull request by sbwsg**
# Changes

Catlin offers a great framework for linting and validating
whether Catalog entries contain any footguns when they're
submitted.

This commit adds a new validation for Steps that warns against
using params directly in script blocks and a unit test to exercise
the new warning. We advise against doing this in [our catalog docs
here](https://github.com/tektoncd/catalog/blob/main/recommendations.md#dont-use-interpolation-in-scripts-or-string-arguments) so it seems like a good candidate for catching with a lint.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md)
for more details._

### CODE REVIEW

File: catlin/pkg/validator/task_validator.go [hunk 1]
```diff
@@ -112,6 +113,17 @@ func (t *taskValidator) validateStep(s v1beta1.Step) Result {
 		result.Warn("Step %q uses secret as environment variables. Prefer using secrets as files over secrets as environment variables", step)
 	}
 
+	if s.Script != "" {
+		expr, _ := extractExpressionFromString(s.Script, "params")
+		if expr != "" {
+			result.Warn(
+				"Step %q references %q directly from its script block. Consider putting the param into an environment variable of the Step and accessing that environment variable in your script instead.",
```
Comment (jerop) [inline catlin/pkg/validator/task_validator.go:120]: I wonder if it's worthwhile including the reasoning for this warning, such as:

```suggestion
				"Step %q references %q directly from its script block. For reliability and security, consider putting the param into an environment variable of the Step and accessing that environment variable in your script instead.",
```

**Review by jerop — APPROVED** [review]
Thank you @sbwsg 🎉

**Comment by sm43** [comment]
lgtm

**Review by vdemeester — APPROVED** [review]
/lgtm

**Pull request merged**